docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+
+The library doesn't depend on the Office JavaScript APIs and it can be used with any JavaScript unit testing framework, including the following among others:
+
+- [Jest](https://jestjs.io)
+- [Mocha](https://mochajs.org/)
+- [Storybook](https://storybook.js.org/docs/react/workflows/unit-testing)
+- [Jasmine](https://jasmine.github.io/)
+
+The examples in this article use the Jest framework. There are examples using the Mocha framework and samples written in TypeScript at [the library's home page](https://github.com/OfficeDev/Office-Addin-Scripts/tree/master/packages/office-addin-mock).
+
+## Prerequisites
+
+This article assumes that you are familiar with the basic concepts of unit testing and mocking, and that you have some experience with a unit testing framework.
+
+> [!TIP]
+> If you are working with Visual Studio, see the article [Unit testing JavaScript and TypeScript in Visual Studio](/visualstudio/javascript/unit-testing-javascript-with-visual-studio.md).
+
+## Install the tool
+
+To install the library open a command prompt, navigate to the root of your add-in project, and then enter the following command:
+
+```command&nbsp;line
+npm install office-addin-mock --save-dev
+```
+
+## Basic usage
+
+1. Require (or import) the library to your test file. The following code is an example:
+
+   ```javascript
+   const OfficeMockObject = require("office-addin-mock");
+   ```
+
+1. Require (or import) the module that contains the add-in function that you want to test. The following is an example that assumes your test file is in a subfolder the folder with your add-in's code files:
+
+   ```javascript
+   const myOfficeAddinFeature = require("../my-office-add-in");
+   ```
+
+1. Create a data object that has the properties and subproperties that you need to mock to test the function. The following is an example of an object that mocks the Excel [Workbook.range.address](/javascript/api/excel/excel.range#address) property and the [Workbook.getSelectedRange](/javascript/api/excel/excel.workbook#getSelectedRange__) method. This isn't the final mock object. Think of it as a seed object that is used by `OfficeMockObject` to create the final mock object.
+
+   ```javascript
+   const mockData = {
+     workbook: {
+       range: {
+         address: "C2:G3",
+       },
+       getSelectedRange: function () {
+         return this.range;
+       },
+     },
+   };
+   ```
+
+1. Pass the data object to the `OfficeMockObject` constructor. Note the following about the returned `OfficeMockObject` object:
+
+   - It is a simplified mock of an [OfficeExtension.ClientRequestContext](/javascript/api/office/officeextension.clientrequestcontext) object. (It is also possible to mock a host object, such as [Excel](/javascript/api/excel) or [Word](/javascript/api/word). For an example, see [Mocking a host object](#mocking-a-host-object).)
+   - The mock object has all the members of the data object and also has mock `load()` and `sync()` methods.
```

Comment:
I found this wording a little weird. Can we do: "and also has mock implementations of `load()` and `sync()` methods." or "and also has already implemented the 'load()' and 'sync()' methods." or something similar?